src/JestProcessManagement/JestProcess.ts

```diff
@@ -0,0 +1,66 @@
+import { Runner, ProjectWorkspace } from 'jest-editor-support'
+
+export class JestProcess {
+  private runner: Runner
+  private exited: boolean
+  private projectWorkspace: ProjectWorkspace
+  public keepAlive: boolean
+  public onExitCallback: Function
+  public watchMode: boolean
+
+  private startRunner() {
+    this.exited = false
```

Comment:
you might encounter a race condition here

Reply:
`startRunner` is a private method. It is only called in two circumstances:

1. When the `JestProcess` instance is created (`startRunner` is called in the constructor).
2. From method `restart`, which is also private and called only after the process already exited. Note that `restart` removes all the handlers from the underlying `Runner` before calling `startRunner`.

So, no race condition as far as I can see, but please elaborate if you thought about something else.

Reply:
if `this.exited` is only used by the closure below, just make it a local variable, it will be captured by the closure.

Reply:
It is related to the other comment below. Making it part of the closure would actually indicate that we are protecting it from something at that would be misleading.

Reply:
the code is cleaner when there is no state, or at least we should keep state as tight as possible, definitely avoid unnecessary state beyond its context. 

Suggested change:
```
private startRunner() {
    let exited = false // here
    this.runner = new Runner(this.projectWorkspace)

    this.runner.start(this.watchMode)

    this.runner.on('debuggerProcessExit', () => {
      if (!exited) { //here
        exited = true // here
        this.onExitCallback(this)
        if (this.keepAlive) {
          this.restart()
        }
      }
    })
  }
```

Reply:
Ok, the argument about keeping the state to the minimum is speaking to me. I will add that.